src/js/utils.js

```diff
@@ -0,0 +1,53 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.d3");
+
+    /**
+     * Convert jQuery DOM elements to D3 DOM elements.
+     * @param domElem - a jQuery DOM element or an array of jQuery DOM elements
+     */
+    gpii.d3.jQueryToD3 = function (elem) {
+        return d3.selectAll(elem.toArray());
+    };
+
+    gpii.d3.addD3Listeners = function (jQueryElem, eventName, listener, that) {
+        var d3Elem = gpii.d3.jQueryToD3(jQueryElem);
+        d3Elem.on(eventName, function (data, i) {
+            fluid.invokeGlobalFunction(listener, [data, i, that]);
+        });
+    };
+
+    /**
+     * Return a D3 color scale based on user supplied colors or the d3.scale.category10() defaults
+     * d3.scale.category10() constructs a new ordinal scale with a range of ten categorical colors:
+     * @param an array of color codes or undefined
+     * @return - a D3 color scale
+     */
+    gpii.d3.getColorScale = function (colors) {
+        return colors ? d3.scale.ordinal().range(colors) : d3.scale.category10();
+    };
```

Comment:
@jobara and I talked more about moving these functions. gpii.d3.getColorScale() is currently used by pie.js, legend.js and pieChart.js. Most functions in this file is D3 related except gpii.isCssClass(). We were thinking to move gpii.isCssClass() into d3ViewComponent.js, then we can give this file a more specific name to indicate things defined in here are D3 related. I will start a conversation in the channel to consult for a proper name.